custom_components/cz_energy_spot_prices/coordinator.py

```diff
@@ -206,9 +206,48 @@ def __init__(
 
         for block in config.all_cheapest_blocks():
             if config.cheapest_blocks_cross_midnight and block is not None:
-                intervals_for_cheapest = self._today_tomorrow_by_dt
+                # Cross-midnight mode enabled for multi-hour blocks
+                if self._tomorrow_day is not None:
+                    # Tomorrow data IS available → filter to FUTURE today + ALL tomorrow
```

Comment:
I'm not sure if filtering the FUTURE today intervals is the correct approach correct. If there are two really cheap blocks today (lets say 02:00 - 03:00 and then 22:00 - 23:00), wouldn't this activate the one hour block for both intervals?